hugegraph-api/src/main/java/com/baidu/hugegraph/auth/StandardAuthenticator.java

```diff
@@ -46,28 +46,33 @@ private HugeGraph graph() {
         return this.graph;
     }
 
-    private void initAdminUser() throws Exception {
-        this.initAdminUser(this.inputPassword());
-
+    private void initAdminUser(AuthManager authManager) throws Exception {
+        if (requireInitAdminUser(authManager)) {
+            this.initAdminUser(this.inputPassword());
+        }
         this.graph.close();
     }
 
-    public void initAdminUser(String password) throws Exception {
+    public void initAdminUser(String password) {
         // Not allowed to call by non main thread
         String caller = Thread.currentThread().getName();
         E.checkState(caller.equals("main"), "Invalid caller '%s'", caller);
 
         AuthManager authManager = this.graph().hugegraph().authManager();
         // Only init user when local mode and user has not been initialized
-        if (StandardAuthManager.isLocal(authManager) &&
-            authManager.findUser(HugeAuthenticator.USER_ADMIN) == null) {
+        if (requireInitAdminUser(authManager)) {
             HugeUser admin = new HugeUser(HugeAuthenticator.USER_ADMIN);
             admin.password(StringEncoding.hashPassword(password));
             admin.creator(HugeAuthenticator.USER_SYSTEM);
             authManager.createUser(admin);
         }
     }
 
+    private boolean requireInitAdminUser(AuthManager authManager) {
+        return StandardAuthManager.isLocal(authManager) &&
```

Comment:
seems can get authManager = this.graph().hugegraph().authManager();

Reply:
shall we use this `this.authManager()` directly?
```java
    @Override
    public AuthManager authManager() {
        return this.graph().authManager();
    }
```